Assets/Scripts/LoudObjects/LoudObject.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LoudObject : MonoBehaviour
+{
+    public float thrust;
+    public Vector3 direction;
+    public float dropRadius;
+    public float innerDropRadius;
+    public float loudObjectRadius;
+    private float distance;
+    private Rigidbody rb;
+
+    public float minShake;
+    public float maxShake;
+    Renderer[] rends;
+    
+
+    void Start()
+    {
+        rb = GetComponent<Rigidbody>();
+        rends = GetComponents<Renderer> ();
+    }
+
+    void Update()
+    {
+        distance = Vector3.Distance(transform.position, GameManager.Instance.GetPlayerTransform().position);
+        if (distance <= innerDropRadius)
+        {
+            rb.AddForce(direction * thrust);
+            rb.useGravity = true;
+            NotifyChasingNurse();
+        } 
+        else if (distance <= (innerDropRadius + 3))
+        {
+            Shake(maxShake);
+        } 
+        else if (distance <= dropRadius)
+        {
+            float dynamicShake = Mathf.Lerp(maxShake, minShake, Mathf.Clamp(distance/5.0f, 0f, 1f));
+            Shake(dynamicShake);
+        } 
+        else 
+        {
+            Shake(0f);
+        }
+    }
+
+    void NotifyChasingNurse()
+    {
+        Collider[] hitColliders = Physics.OverlapSphere(transform.position, loudObjectRadius);

```

Comment:
I'd add a TODO for adding in the 3rd parameter to this function which is a layer mask, so we only check collisions with enemies (I believe they are on a separate layer?)